src/krook_collisions.jl

```diff
@@ -0,0 +1,134 @@
+            nu_ii = collisions.krook_collision_frequency_prefactor * n * T^(-1.5)
+            @loop_vperp_vpa ivperp ivpa begin
+                pdf_out[ivpa,ivperp,iz,ir,is] -= dt * nu_ii *
+                    (fvec_in.pdf[ivpa,ivperp,iz,ir,is]
+                     - exp(-vpa.grid[ivpa]^2 - vperp.grid[ivperp]^2))
+            end
+        end
+    elseif moments.evolve_ppar
+        # Compared to full-f collision operater, multiply through by vth, remembering pdf
+        # is already multiplied by vth, and grid is already normalized by vth
+        @loop_s_r_z is ir iz begin
+            n = fvec_in.density[iz,ir,is]
+            vth = moments.charged.vth[iz,ir,is]
+            T = vth^2
+            nu_ii = collisions.krook_collision_frequency_prefactor * n * T^(-1.5)
+            @loop_vperp_vpa ivperp ivpa begin
+                pdf_out[ivpa,ivperp,iz,ir,is] -= dt * nu_ii *
+                    (fvec_in.pdf[ivpa,ivperp,iz,ir,is]
+                     - exp(-((vpa.grid[ivpa] - fvec_in.upar[iz,ir,is])/vth)^2
+                           - (vperp.grid[ivperp]/vth)^2))
+            end
+        end
+    elseif moments.evolve_upar
+        # Compared to evolve_density version, grid is already shifted by upar
+        @loop_s_r_z is ir iz begin
+            n = fvec_in.density[iz,ir,is]
+            vth = moments.charged.vth[iz,ir,is]
+            T = vth^2
+            nu_ii = collisions.krook_collision_frequency_prefactor * n * T^(-1.5)
+            @loop_vperp_vpa ivperp ivpa begin
+                pdf_out[ivpa,ivperp,iz,ir,is] -= dt * nu_ii *
+                    (fvec_in.pdf[ivpa,ivperp,iz,ir,is]
+                     - 1.0 / vth * exp(-(vpa.grid[ivpa] / vth)^2
+                                       - (vperp.grid[ivperp] / vth)^2))
+            end
+        end
+    elseif moments.evolve_density
+        # Compared to full-f collision operater, divide through by density, remembering
+        # that pdf is already normalized by density
+        @loop_s_r_z is ir iz begin
+            n = fvec_in.density[iz,ir,is]
+            vth = moments.charged.vth[iz,ir,is]
+            T = vth^2
+            nu_ii = collisions.krook_collision_frequency_prefactor * n * T^(-1.5)
+            @loop_vperp_vpa ivperp ivpa begin
+                pdf_out[ivpa,ivperp,iz,ir,is] -= dt * nu_ii *
+                (fvec_in.pdf[ivpa,ivperp,iz,ir,is]
+                 - 1.0 / vth
+                 * exp(-((vpa.grid[ivpa] - fvec_in.upar[iz,ir,is]) / vth)^2
+                           - (vperp.grid[ivperp]/vth)^2))
+            end
+        end
+    else
+        @loop_s_r_z is ir iz begin
+            n = fvec_in.density[iz,ir,is]
+            vth = moments.charged.vth[iz,ir,is]
+            T = vth^2
+            if vperp.n == 1
+                vth_prefactor = 1.0 / vth
+            else
+                vth_prefactor = 1.0 / vth^3
+            end
+            nu_ii = collisions.krook_collision_frequency_prefactor * n * T^(-1.5)
```

Comment:
This choice of prefactor is probably motivated by the physical form of Fokker-Planck collision frequency, but it is not obvious to me that we would actually want this feature. Isn't it easier to understand the ad-hoc feature that we add if it doesn't have a nonlinear collision frequency?

Reply:
I think it makes sense to have both options (e.g. Gkeyll does that https://gkeyll.readthedocs.io/en/latest/gkyl/App/Collisions/collisionModels.html). If you want to add an option for constant collision frequency I'm happy for that to be there. I wanted to have a density/temperature dependent collision frequency because in a setup where physically the collisionality changes significantly (e.g. due to a large temperature drop between midplane and target) I don't want to have unphysically strong collisions upstream, but want the collisions near the target to be strong enough to Maxwellian-ise the plasma.